feature-libs/product-configurator/rulebased/cpq/cpq-configurator-overview-normalizer.ts

```diff
@@ -0,0 +1,165 @@
+        .pipe(take(1))
+        .subscribe(
+          (generalText) => (groupOverview.groupDescription = generalText)
+        );
+    }
+    return groupOverview;
+  }
+
+  protected convertAttribute(
+    attr: Cpq.Attribute,
+    currency: string
+  ): Configurator.AttributeOverview[] {
+    const attributeOverviewType: Configurator.AttributeOverviewType =
+      attr?.values && this.cpqUtilitiesService.hasAnyProducts(attr?.values)
+        ? Configurator.AttributeOverviewType.BUNDLE
+        : Configurator.AttributeOverviewType.GENERAL;
+    const ovAttr: Configurator.AttributeOverview[] = [];
+    this.convertAttributeValue(attr, currency).forEach((ovValue, index) => {
+      ovAttr.push({
+        ...ovValue,
+        type: attributeOverviewType,
+      });
+      ovAttr[index].attribute = this.cpqUtilitiesService.retrieveAttributeLabel(
+        attr
+      );
+    });
+    return ovAttr;
+  }
+
+  protected convertAttributeValue(
+    attr: Cpq.Attribute,
+    currency: string
+  ): Configurator.AttributeOverview[] {
+    const ovValues: Configurator.AttributeOverview[] = [];
+    switch (attr.displayAs) {
+      case Cpq.DisplayAs.INPUT:
+        if (attr?.dataType === Cpq.DataType.INPUT_STRING) {
+          if (attr.userInput && attr.userInput.length > 0) {
+            ovValues.push(this.extractOvValueUserInput(attr, currency));
+          }
+        } else {
+          ovValues.push({ attribute: undefined, value: 'NOT_IMPLEMENTED' });
+        }
+        break;
+      case Cpq.DisplayAs.RADIO_BUTTON:
+      case Cpq.DisplayAs.DROPDOWN:
+        const selectedValue = attr.values?.find(
+          (val) => val.selected && val.paV_ID !== NO_OPTION_SELECTED
+        );
+        if (selectedValue) {
+          ovValues.push(this.extractOvValue(selectedValue, attr, currency));
+        }
+        break;
+      case Cpq.DisplayAs.CHECK_BOX:
+        attr.values
+          ?.filter((val) => val.selected)
+          ?.forEach((valueSelected) => {
+            ovValues.push(this.extractOvValue(valueSelected, attr, currency));
+          });
+        break;
+      default:
+        ovValues.push({ attribute: undefined, value: 'NOT_IMPLEMENTED' });
+    }
+    return ovValues;
+  }
+
+  protected extractOvValue(
+    valueSelected: Cpq.Value,
+    attr: Cpq.Attribute,
+    currency: string
+  ): Configurator.AttributeOverview {
+    const ovValue: Configurator.AttributeOverview = {
+      attribute: undefined,
+      value: valueSelected.valueDisplay,
+      productCode: valueSelected.productSystemId,
+      quantity: this.cpqUtilitiesService.prepareQuantity(valueSelected, attr),
+      valuePrice: this.cpqUtilitiesService.prepareValuePrice(
+        valueSelected,
+        currency
+      ),
+    };
+    ovValue.valuePriceTotal = this.cpqUtilitiesService.calculateValuePriceTotal(
+      ovValue.quantity,
+      ovValue.valuePrice
+    );
+    return ovValue;
+  }
+
+  protected extractOvValueUserInput(
```

Comment:
For the clarity of the public API, let's avoid custom abbreviations like `Ov` in the methods names.

Reply:
Pick!

Reply:
Done!